### ISSUE TITLE
Terraform State Doesn't Refresh

### ISSUE THREAD

**Issue by luckerby** [hashicorp/terraform#20091]
### Terraform Version
```
Terraform v0.11.11
```

As part of provisioning a number of VMs using the Vsphere provider, I'm invoking an external tool each time for each VM. The tool sends back the IP, netmask and gateway to be used for each VM (this information is retrieved by the tool itself from an IPAM solution). Since I develop the external tool gradually, until recently, the netmask was hardcoded as seen below.

```
resource "vsphere_virtual_machine" "vm" {
...
   clone {
      customize {
          network_interface {
              # Use * to iterate through all the elements of NetworkObtainedData
              ipv4_address = "${data.external.NetworkObtainedData.*.result.ip[count.index]}"
              ipv4_netmask = 23
...
```
However recently, progress has been made which also allows the netmask to be returned. Yet due to a bug, instead of returning the netmask as a number of bits (eg 23), it returned the mask in the dotted notation (eg 255.255.254.0). terraform rightfully complained:

```
C:\Users\malbert\Desktop\Excel2VM>terraform plan -out deployVM

Error: vsphere_virtual_machine.vm[0]: clone.0.customize.0.network_interface.0.ipv4_netmask: cannot parse '' as int: strconv.ParseInt: parsing "255.255.254.0": invalid syntax

Error: vsphere_virtual_machine.vm[1]: clone.0.customize.0.network_interface.0.ipv4_netmask: cannot parse '' as int: strconv.ParseInt: parsing "255.255.254.0": invalid syntax
```

The trouble is that after the bug was fixed - and the tool returns the netmask as a number of bits - terraform still complains. It appears as if the state is not being refreshed. Even manually invoking `terraform refresh` results in the same error above thrown.

### Expected Behavior
I would have expected at least a part of the output below...
```
C:\Users\malbert\Desktop\Excel2VM>terraform plan -out deployVM
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.external.IPregexMatchPattern: Refreshing state...
data.external.NetworkObtainedData[0]: Refreshing state...
data.external.NetworkObtainedData[1]: Refreshing state...
data.vsphere_datacenter.dc: Refreshing state...
data.vsphere_datastore.datastore: Refreshing state...
data.vsphere_network.network: Refreshing state...
data.vsphere_virtual_machine.template: Refreshing state...
data.vsphere_resource_pool.pool: Refreshing state...
vsphere_virtual_machine.vm[1]: Refreshing state... (ID: 4227a1a3-bf73-0024-e240-5c925147e96a)
vsphere_virtual_machine.vm[0]: Refreshing state... (ID: 42275cfa-4dde-3c2d-4ece-44edd599ad9d)
```

### Actual Behavior
...instead either a `plan` or `refresh` results in the immediate errors seen below.
```
C:\Users\malbert\Desktop\Excel2VM>terraform refresh

Error: vsphere_virtual_machine.vm[0]: clone.0.customize.0.network_interface.0.ipv4_netmask: cannot parse '' as int: strconv.ParseInt: parsing "255.255.254.0": invalid syntax

Error: vsphere_virtual_machine.vm[1]: clone.0.customize.0.network_interface.0.ipv4_netmask: cannot parse '' as int: strconv.ParseInt: parsing "255.255.254.0": invalid syntax
```

**Comment by luckerby** [comment]
Testing further shows that at this point no command can be issued, including `terraform destroy`. 

I went ahead and modified the .tfstate to manually correct the references from `255.255.254.0` to `23`. However the same error is thrown. I've also powered off 1 VM out of the 2, just to be extra sure that the guests themselves aren't polled for values, and this would create the error, yet still no change in outcome. 

I've enabled tracing by setting the `TF_LOG` to `trace`, and placed the resulting output for `terraform refresh` [here](https://gist.github.com/luckerby/3b8e430da5bfc6baed0af769b9488edc).
